src/hyperfine/benchmark.rs

```diff
@@ -201,6 +201,26 @@ fn run_cleanup_command(
     run_intermediate_command(shell, command, show_output, error_output)
 }
 
+#[cfg(unix)]
+fn extract_exit_code(status: ExitStatus) -> i32 {
+    use std::os::unix::process::ExitStatusExt;
+
+    /* From the ExitStatus::code documentation:
+       "On Unix, this will return None if the process was terminated by a signal."
+       In that case, ExitStatusExt::signal should never return None.
+    */
+    status.code().unwrap_or_else(|| status.signal().unwrap())
+}
+
+#[cfg(not(unix))]
+fn extract_exit_code(status: ExitStatus) -> i32 {
+    /* From the ExitStatus::code documentation:
+       "On Unix, this will return None if the process was terminated by a signal."
+       On the other configurations, ExitStatus::code should never return None.
+    */
```

Comment:
Ok, but I don't really see a guarantee for the latter part ("On the other configurations, ExitStatus::code should never return None.") in the documentation.

I think it might be best to simply return an `Option<i32>` from this function and later (in the export) convert exit codes to a string that either contains the exit code or an empty string. What do you think?

Reply:
@JordiChauzi Are you interested in finishing this? Otherwise, that's also fine! We can find someone else then.

Reply:
Sorry about the delay. Last month was a little busy and I kind of forgot about this PR.
Anyway, I will try and implement the changes in the next days.
You are right, there is no real guarantee that a configuration may return `None` from the `ExitStatus::code` method (this is currently not the case but again, this is not a guarantee). Printing an empty string in that case makes sense. Do we also want to store `None` instead of unwrapping when the `ExitStatus::signal` method returns `None`?

Reply:
The default behavior for `serde` and outputting `json` is to serialize `None` as `null`. This will create outputs that look like this:
```json
...
"exit_codes": [
        0,
        0,
        0,
        1,
        null,
        null,
        42,
        0,
        0,
        0
      ]
...
```

To me, it looks "good enough". If you still want the empty string behavior tell me so. If this is the case, I think I will have to add a custom serialize method pour the `exit_codes` field?

Reply:
No, I like the `null` solution better. You are right. Thanks!